### PR TITLE
refactor: introduce monorepo to Generator Repo

### PR DESCRIPTION
**Pull request by ayushnau**
Changes Added. 
1. Turborepo Added. 
2. Converted the generator repo to app,  inside the turborepo. 
3. Added the required scripts of the turborepo to the root package.json file. 

**Related issue(s)**

Resolves https://github.com/asyncapi/generator/issues/1044

### CODE REVIEW

**Comment by ayushnau** [comment]
@derberg can you check this now the test are working fine. 
image
your advise worked.

**Comment by aeworxet** [comment]
@asyncapi/bounty_team

**Comment by AyushNautiyalDeveloper** [comment]
image

@derberg  please check added the required lint change.

**Comment by derberg** [comment]
@ayushnau @AyushNautiyalDeveloper there are errors https://github.com/asyncapi/generator/actions/runs/9207081971/job/25326746738?pr=1213

**Comment by ayushnau** [comment]
let me check though i have fixed these in the local.

**Comment by derberg** [comment]
@ayushnau I see linter complains on files from `/home/runner/work/generator/generator/apps/generator/test/test-templates/react-template/__transpiled/test-file.md.js` but it should not as these should be ignored

so basically make sure that eslint ignores `/apps/generator/test/test-templates/react-template/__transpiled`

**Comment by derberg** [comment]
I see it is there already in `.eslintignore` but you most probably need to update lint scripts to point to ignore file in root

**Comment by ayushnau** [comment]
@derberg I have already pushed the changes for eslintignore can you check.

**Comment by derberg** [comment]
to fix https://github.com/asyncapi/generator/actions/runs/9211948474/job/25347170425?pr=1213 you need to update https://github.com/asyncapi/generator/blob/master/.github/workflows/pr-testing-with-test-project.yml#L21 probably to `./apps/generator/test/test-project`

**Comment by ayushnau** [comment]
> to fix https://github.com/asyncapi/generator/actions/runs/9211948474/job/25347170425?pr=1213 you need to update https://github.com/asyncapi/generator/blob/master/.github/workflows/pr-testing-with-test-project.yml#L21 probably to `./apps/generator/test/test-project`

sure will do this.

File: apps/generator/package.json [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    "test": "npm run test:unit && npm run test:integration && npm run test:cli",
+    "test:unit": "jest --coverage --testPathIgnorePatterns=integration --testPathIgnorePatterns=test-project",
+    "test:dev": "npm run test:unit -- --watchAll",
+    "test:integration": "npm run test:cleanup && jest --testPathPattern=integration --modulePathIgnorePatterns='./__mocks__'",
+    "test:integration:update": "jest --updateSnapshot --testPathPattern=integration --modulePathIgnorePatterns='./__mocks__'",
+    "test:cli": "node cli.js ./test/docs/dummy.yml ./test/test-templates/react-template -o test/output --force-write --debug && test -e test/output/test-file.md",
+    "test:cleanup": "rimraf \"test/temp\"",
+    "docs": "jsdoc2md --partial docs/jsdoc2md-handlebars/custom-sig-name.hbs docs/jsdoc2md-handlebars/main.hbs docs/jsdoc2md-handlebars/docs.hbs docs/jsdoc2md-handlebars/header.hbs docs/jsdoc2md-handlebars/defaultvalue.hbs docs/jsdoc2md-handlebars/link.hbs docs/jsdoc2md-handlebars/params-table.hbs --files lib/generator.js > docs/api.md",
+    "docker:build": "docker build -t asyncapi/generator:latest .",
+    "lint": "eslint --max-warnings 0 --config ../../.eslintrc .",
```
Comment (derberg) [inline apps/generator/package.json:24]:
```suggestion
    "lint": "eslint --max-warnings 0 --config ../../.eslintrc --ignore-path ../../.eslintignore .",
```

linting still failing, maybe in CI it still needs proper explicit config for eslintignore

**Comment by ayushnau** [comment]
image
image
@derberg please check updated and added the required chagnes for the release.

**Pull request merged**